src/librustdoc/html/static/js/search.js

```diff
@@ -167,33 +484,141 @@ window.initSearch = function(rawSearchIndex) {
     }
 
     /**
-     * Executes the query and returns a list of results for each results tab.
-     * @param  {Object}        query          - The user query
-     * @param  {Array<string>} searchWords    - The list of search words to query against
-     * @param  {string}        [filterCrates] - Crate to search in
-     * @return {{
-     *   in_args: Array<?>,
-     *   returned: Array<?>,
-     *   others: Array<?>
-     * }}
+     * Parses the query.
+     *
+     * The supported syntax by this parser is as follow:
+     *
+     * ident = *(ALPHA / DIGIT / "_")
+     * path = ident *(DOUBLE-COLON ident)
+     * arg = path [generics]
+     * arg-without-generic = path
+     * nonempty-arg-list = arg *WS *(COMMA/WS *WS arg)
+     * nonempty-arg-list-without-generics = arg-without-generic *WS *(COMMA/WS *WS arg
```

Comment:
The actual parser accepts trailing commas, as in `a,b,`. We can change these to match.

```suggestion
     * type-sep = COMMA/WS *(COMMA/WS)
     * nonempty-arg-list = arg *(type-sep arg) *(COMMA/WS)
     * nonempty-arg-list-without-generics = arg-without-generic *(type-sep arg-without-generic) *(COMMA/WS)
```